#20 update event handler

nmohan101 opened the pull request:
Changes:

1. Renamed Application directory to bin 
2. Added a new directory lib and moved asynctimer.py, getip.py into it. Also added log.py and sql.py to this
3. Updated client.py, server.py, event_handler.py and capture.py import log instead of init in individual files
4. Moved SQL class into its own file (sql.py) from event_handler.py. Updated event_handler so its easier to add new tables if needed in the future. 
5. Updated install script to accommodate new directories. 
6. Update client.py to not error out if PiCamera is not physically installed. (Allow Sim mode)

nmohan101 commented on bin/capture.py:
```diff
@@ -16,6 +16,12 @@
 import logging
```
This can be removed as its repetitive. from log import log takes care of this.

nmohan101 commented on lib/log.py:
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+
+"""log.py: This library faciliates the logging throughput the application. 
```
typo.

codemonkey111 commented on Application/event_handler.py:
```diff
@@ -27,15 +27,15 @@
 #                   Constants                       #
 #---------------------------------------------------#
 UID = pwd.getpwuid(os.getuid()).pw_uid
-DBNAME = "/srv/PhotoNetwork/PhotoNetwork.db"
```
What's up here?  Why did this get removed?  Is it defined in the 'Log' file?

nmohan101 replied:
Yes. I moved this whole class into a separate file (lib/sql.py)

codemonkey111 commented on Application/event_handler.py:
```diff
@@ -93,112 +93,12 @@ def _create_struct(self):
             all_index += db_index_data
         return db_data, all_index, sql
 
-class SQL(object):
```
This whole section, what is it? why remove it? What did it do and does it do now?

nmohan101 replied:
moved to sql.py. Its the class that is used to read/write to db. Create the db. Basically anything to do with the database is done through this class.

codemonkey111 commented on Application/event_handler.py:
```diff
@@ -317,25 +217,9 @@ def read_fifo():
     parser.add_argument('-v', '--verbosity', action = "store_true",  
                         help = "Enter -v for verbosity")
     args = parser.parse_args()
-    
```
This is the section that you moved to it's own file, correct?

nmohan101 replied:
yes.

nmohan101 commented:
Latest commit: 
- Fixed duplication of "import logging" in all files. 
- Fixed typos
- Captures were being overwritten as the file name had only data/time and if 2 captures were taken in the same second file would be over written. (Now each capture has a unique number)
- Updated path for capture.py in srv_capture.c < Eventually this path should come from a config file
- Updated multi-listen function. (rx_data should have been self.rx_data). Removed eval function so now only a string needs to be passed to the multi_listen function.
